extends Control

@onready var fondo := $His1        # Tu Sprite2D
@onready var texto := $LabelTexto
@onready var boton := $BotonContinuar

# 游닆 Tres partes de historia (texto e imagen)
var escenas = [
	{
		"imagen": "res://assets/images/his1.png",
		"texto": """
Hace mucho tiempo, en una tierra olvidada, una Bolivia post apocal칤ptica, dominada por almas
malditas, pose칤das por un ente maligno, exist칤an dos hermanos, Luis y Juan, quienes a pesar de 
no tener mucho y sobrevivir cada d칤a, se las ingeniaban para vivir c칩modamente.
"""
	},
	{
		"imagen": "res://assets/images/his2.png",
		"texto": """
Hasta que un d칤a, Juan encontr칩 un perrito que parec칤a lastimado.En su inocencia, decidi칩 seguirlo,
 sin saber que detr치s de aquel animal estaba aquel ente responsable del dolor de muchas personas: 
el mism칤simo Canibaba Kilure, esperando pacientemente para disfrutar otra de sus v칤ctimas.
"""
	},
	{
		"imagen": "res://assets/images/his3.png",
		"texto": """
Luis, al ver que su hermano hab칤a desaparecido, entr칩 en un miedo agonizante. Por la desesperaci칩n, 
dej칩 su refugio y se llev칩 lo poco que ten칤a. Con un machete en mano, decidi칩 ir en busca 
de su mejor amigo, su peque침o hermano, aun sabiendo que se enfrentar칤a a cientos de almas malditas. 
Aun as칤, Luis decidi칩 ir... sin darse cuenta de que aquellos ojos del bosque ya lo estaban observando.
"""
	}
]

var indice := 0
var escribiendo := false

func _ready():
	var stream = load("res://assets/Music/musica/fondo-bosque.ogg")
	var audio = AudioStreamPlayer.new()
	add_child(audio)
	audio.stream = stream
	audio.autoplay = false
	audio.volume_db = -6
	audio.play()

	boton.pressed.connect(_on_boton_continuar_pressed)
	_mostrar_historia_actual()
# 游뿼 Mostrar imagen y texto con efecto m치quina de escribir
func _mostrar_historia_actual():
	var escena_actual = escenas[indice]

	# Cambia la imagen (Sprite2D)
	fondo.texture = load(escena_actual["imagen"])
	fondo.scale = Vector2(0.75, 0.64)  # ajusta al tama침o que se vea bien en tu pantalla

	# Muestra el texto actual
	texto.text = escena_actual["texto"]
	texto.visible_characters = 0
	escribiendo = true
	boton.disabled = true

	var timer := Timer.new()
	timer.wait_time = 0.03
	timer.one_shot = false
	add_child(timer)
	timer.timeout.connect(func():
		texto.visible_characters += 1
		if texto.visible_characters >= texto.text.length():
			timer.queue_free()
			escribiendo = false
			boton.disabled = false
	)
	timer.start()

# 游녤 Avanzar entre las historias o ir al juego
func _on_boton_continuar_pressed():
	if escribiendo:
		# Si a칰n se escribe, muestra todo el texto al instante
		texto.visible_characters = texto.text.length()
		escribiendo = false
		boton.disabled = false
		return

	indice += 1
	if indice < escenas.size():
		_mostrar_historia_actual()
	else:
		get_tree().change_scene_to_file("res://ecenas/Ecenas/primer_ecena.tscn")
